### PR TITLE
Schema guide descriptions for root-level keys abstract through message

### PR DESCRIPTION
**Pull request by jspaaks**
**Related issues**

Refs: #206

**Describe the changes made in this pull request**

Added more descriptions for the root level keys `abstract` through `messsage`. Not fully done if you take into account the deeper levels of descriptions, but probably a good idea to try to merge now in order to avoid merge conflicts later.

Extra: updated the config for markdown link checker a bit, found an issue #243 

**Instructions to review the pull request**

Look at schema-guide.md, verify that the link checker github action is green

### CODE REVIEW

File: schema-guide.md
Comment (jspaaks) [inline schema-guide.md:1635]:
```suggestion
The person's name particle, e.g., a nobiliary particle or a preposition meaning 'of' or 'from' (for example 'von' in 'Alexander von Humboldt').
```

File: schema-guide.md
Comment (jspaaks) [inline schema-guide.md:1649]:
```suggestion
The person's name-suffix, e.g. 'Jr.' for Sammy Davis Jr. or 'III' for Frank Edwin Wright III.
```

File: schema-guide.md
Comment (sdruskat) [inline schema-guide.md:303]:
```suggestion
 - a third keyword
```

Just to make this complete and in line with the two examples above, taken this is meant to show that you can use a string with spaces here.

File: schema-guide.md
Comment (sdruskat) [inline schema-guide.md:363]:
```suggestion
message: If you use this dataset, please cite it using the metadata from this file.
```

[CED](https://dictionary.cambridge.org/dictionary/english/dataset) says one word, although [APA Style Blog](https://blog.apastyle.org/apastyle/2012/07/data-is-or-data-are.html) says differently (but who are they in the world of lexicographers!), so I asked the world corpus Google Search which gives 64M hits for `"data set"`and 125M for `"dataset"`. Clear result if you ask me.

File: schema-guide.md
Comment (sdruskat) [inline schema-guide.md:366]:
```suggestion
message: Please cite this dataset using these metadata.
```

See https://github.com/citation-file-format/citation-file-format/pull/242/files#r676562871

File: schema-guide.md
Comment (sdruskat) [inline schema-guide.md:369]:
```suggestion
message: Please cite this dataset using the metadata from 'preferred-citation'.
```

See https://github.com/citation-file-format/citation-file-format/pull/242/files#r676562871

File: schema-guide.md
Comment (sdruskat) [inline schema-guide.md:254]:
```suggestion
The DOI of the software or dataset. This notation is most useful when there is just one DOI you want to include. In
```
See https://github.com/citation-file-format/citation-file-format/pull/242/files#r676562871

File: schema-guide.md
Comment (sdruskat) [inline schema-guide.md:764]:
```suggestion
    fax: +12-3456-7890
```

Just an OCD alert, the `+` replaces the intl `00` "Verkehrsausscheidungsziffer" (there isn't even an English Wikipedia page for telephone country codes 😱, the related Dutch one calls this simply "[Landnummer](https://nl.wikipedia.org/wiki/Landnummer)" I think), and there are countries without the `0` prefix, so <wince> can we not use this here?

File: schema-guide.md
Comment (sdruskat) [inline schema-guide.md:832]:
```suggestion
    tel: +12-345-6789098
```

See https://github.com/citation-file-format/citation-file-format/pull/242/files#r676568214

File: schema-guide.md
Comment (sdruskat) [inline schema-guide.md:865]:
```suggestion
    fax: +12-3456-7890
```

See https://github.com/citation-file-format/citation-file-format/pull/242/files#r676568214

File: schema-guide.md
Comment (sdruskat) [inline schema-guide.md:870]:
```suggestion
    fax: +12-3456-7890
```

https://github.com/citation-file-format/citation-file-format/pull/242/files#r676568214

File: schema-guide.md
Comment (sdruskat) [inline schema-guide.md:3488]:
```suggestion
    tel: +12-345-6789098
```

See https://github.com/citation-file-format/citation-file-format/pull/242/files#r676568214

File: schema-guide.md
Comment (sdruskat) [inline schema-guide.md:3493]:
```suggestion
    tel: +12-345-6789098
```

See https://github.com/citation-file-format/citation-file-format/pull/242/files#r676568214

File: schema-guide.md
Comment (sdruskat) [inline schema-guide.md:1594]:
```suggestion
    fax: +12-3456-7890
```

See https://github.com/citation-file-format/citation-file-format/pull/242/files#r676568214

File: schema-guide.md
Comment (sdruskat) [inline schema-guide.md:1697]:
```suggestion
    tel: +12-345-6789098
```

See https://github.com/citation-file-format/citation-file-format/pull/242/files#r676568214

File: schema-guide.md
Comment (sdruskat) [inline schema-guide.md:920]:
```suggestion
    description: The GitHub release URL of tag 1.1.0.
```

IMHO it should be "locator of" ...

File: schema-guide.md
Comment (sdruskat) [inline schema-guide.md:923]:
```suggestion
    description: The GitHub release URL to the commit of tag 1.1.0.
```

See https://github.com/citation-file-format/citation-file-format/pull/242/files#r676571683

File: schema-guide.md
Comment (sdruskat) [inline schema-guide.md:965]: No way around this big list I guess. We could contract into a comma-separated list to save screen space, but this would probably make it less usable, what do you think?

File: schema-guide.md
Comment (sdruskat) [inline schema-guide.md:1514]:
```suggestion
- type: [`definitions.alias`](#definitionsalias)
```

Remove duplicate `- type: ` ...

File: schema-guide.md
Comment (sdruskat) [inline schema-guide.md:3523]:
```suggestion
  - name: The Research Software Project
```
While this would've been valid YAML AFAIK, put on one line to save space (and let people figure out their YAML by reading *their* specs).

**Review by sdruskat — APPROVED** [review]
Thanks!
Happy to merge this now once the (mostly OCD-related) comments are resolved.
I'll rebase any further work on `1.2.0` once it's merged...

File: schema-guide.md
Comment (jspaaks) [reply schema-guide.md:363]: good enough for me :)

**Review by jspaaks — COMMENTED** [review]
somehow github thinks I'm reviewing my own PR, just ignore this comment

File: schema-guide.md
Comment (jspaaks) [reply schema-guide.md:965]: yeah there isnt really a good way of dealing with this. Maybe a table could work, but I think I'm right in saying you cant have a table nested in a list. Alternatively, we could do a fixed-width font multicolumn print, like what you would see in a terminal, to help people not injure themselves scrolling through the license list.

**Comment by jspaaks** [comment]
>  Happy to merge this now once the (mostly OCD-related) comments are resolved.

Don't worry, that's what makes you so relatable (...to me)

**Comment by jspaaks** [comment]
Thanks for reviewing!

**Pull request merged**
